test/e2e/common.go

```diff
@@ -127,14 +127,23 @@ func (test *e2eTest) DeleteTestNamespace(t *testing.T, namespace string) {
 // WaitForNamespaceDeleted wait until namespace is deleted
 func (test *e2eTest) WaitForNamespaceDeleted(t *testing.T, namespace string) {
 	logger := Logger{}
-	deleted := checkNamespaceDeleted(t, namespace, MaxRetries, logger)
+	deleted := checkNamespace(t, namespace, false, MaxRetries, logger)
 	if !deleted {
 		t.Fatalf("Error deleting namespace %s, timed out", namespace)
 	}
 }
 
+// WaitForNamespaceCreated wait until namespace is created
+func (test *e2eTest) WaitForNamespaceCreated(t *testing.T, namespace string) {
+	logger := Logger{}
+	created := checkNamespace(t, namespace, true, MaxRetries, logger)
+	if !created {
+		t.Fatalf("Error creating namespace %s, timed out", namespace)
+	}
+}
+
 // Private functions
-func checkNamespaceDeleted(t *testing.T, namespace string, maxRetries int, logger Logger) bool {
+func checkNamespace(t *testing.T, namespace string, created bool, maxRetries int, logger Logger) bool {
```

Comment:
I think it would be good to return an error (even in the "timeout" case) so that if something goes wrong we can echo it back to the user and they can try to fix it.

Reply:
But the `t.Fatalf(..)` call will fail the test and the user will see the error...

Reply:
Ignore my comments - I didn't realize this is just for testing - I thought this was "real" code :-)